src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit/Models/Propagated.cpp

```diff
@@ -0,0 +1,675 @@
+{
+    std::cout.precision(2) ;
+    std::cout.setf(std::ios::fixed,std::ios::floatfield) ;
+
+    std::cout << std::left << std::setw(15) << t ;
+    std::cout.precision(8) ;
+    std::cout.setf(std::ios::scientific,std::ios::floatfield) ;
+
+    std::cout << std::internal << std::setw(16) << x[0] << "     " << std::internal << std::setw(16) << x[1] << "     " << std::internal << std::setw(16) << x[2] << "     " << std::internal << std::setw(16) << x[3] << "     " << std::internal << std::setw(16) << x[4] << "     " << std::internal << std::setw(16) << x[5] << std::endl ;
+    std::cout.setf(std::ios::fixed,std::ios::floatfield) ;
+}
+
+State                           Propagated::CalculateNoneStateAt                (   const   State&                  aState,
+                                                                                    const   Derived&                aGravitationalParameter,
+                                                                                    const   Instant&                anInstant,
+                                                                                    const   Propagated::IntegrationStepperType& anIntegrationStepperType,
+                                                                                    const   Propagated::IntegrationLogType& anIntegrationLogType,
+                                                                                    const   Real&                   aRelativeTolerance,
+                                                                                    const   Real&                   anAbsoluteTolerance                         )
+{
+
+    using ostk::physics::units::Time ;
+    using ostk::physics::units::Derived ;
+    using ostk::physics::units::Angle ;
+    using ostk::physics::time::Duration ;
+    using ostk::physics::coord::Position ;
+    using ostk::physics::coord::Velocity ;
+
+    using ostk::math::obj::Vector3d ;
+
+    const double mu_SI = aGravitationalParameter.in(GravitationalParameterSIUnit) ;
+
+    // Get state related parameters
+    Position currentPosition = aState.getPosition();
+    Velocity currentVelocity = aState.getVelocity();
+
+    Vector3d currentPositionCoords = currentPosition.inUnit(Position::Unit::Meter).accessCoordinates();
+    Vector3d currentVelocityCoords = currentVelocity.inUnit(Velocity::Unit::MeterPerSecond).accessCoordinates();
+
+    Propagated::state_type fullState(6);
+    fullState[0] = currentPositionCoords[0];
+    fullState[1] = currentPositionCoords[1];
+    fullState[2] = currentPositionCoords[2];
+    fullState[3] = currentVelocityCoords[0];
+    fullState[4] = currentVelocityCoords[1];
+    fullState[5] = currentVelocityCoords[2];
+
+    Duration propDuration = anInstant - aState.getInstant() ;
+    double propDurationInSecs = (double)propDuration.inSeconds() ;
+
+    const double timeStep = 5.0; // Default value for now
```

Comment:
move to be configurable as an argument?

Reply:
yes, will change that once I move around my code to the new layout